src/compose/app.ts

```diff
@@ -881,7 +885,7 @@ class AppImpl implements App {
 		// different to a dependency which is in the servicePairs below, as these
 		// are services which are changing). We could have a dependency which is
 		// starting up, but is not yet running.
-		const depInstallingButNotRunning = _.some(targetApp.services, (svc) => {
+		const depInstallingButNotRunning = _.some(this.services, (svc) => {
```

Comment:
Bug fix. Services in the target app are generally configured a `running: true` so dependencies would get started without waiting for the status of the actual container